### PR TITLE
Highlight searched keywords in matches

### LINKED ISSUE #610: IE (and Edge?) clear button in Monitoring page filter input doesn't clear the filter
IE (and Edge) adds a "X" clear button to text inputs once they have content entered in to them. 

screen shot 2016-09-29 at 3 42 01 pm

Clicking said "X" clears the input, but does not remove the filter from the contents displayed below.

screen shot 2016-09-29 at 3 45 00 pm

We should either [disable the "X"](http://stackoverflow.com/questions/13481577/how-can-i-disable-the-clear-button-that-ie10-inserts-into-textboxes) or make the "X" unset the filter. 

Better yet, we should add an "X" for all browsers that clears the filter on any filter we have in the app.

### PR DESCRIPTION
**Pull request by spadgett**
* Update the project list, add to project catalog, monitoring page, and
  events directive to highlighting matching keywords for the as you type
  filter.
* Use KeywordService to search events.
* Add a clear button to the events and monitoring page filter input for
  consistency with other search boxes.

openshift_web_console

Fixes #610

@jwforres @rhamilto @sspeiche @luciddreamz

### CODE REVIEW

**Comment by luciddreamz** [comment]
Makes total sense, love it

**Comment by spadgett** [comment]
@jwforres PTAL

**Comment by jwforres** [comment]
@spadgett needs to be rebased

**Comment by spadgett** [comment]
@jwforres Rebased. Tested the highlighting with event table links:

openshift_web_console

File: app/views/catalog/_image.html [hunk 1]
```diff
@@ -5,16 +5,18 @@
...
         </div>
         <p class="card-pf-badge">Builds source code</p>
         <p>
           <truncate-long-text
+            ng-if="!keywords.length"
             class="project-description"
             content="imageStream | imageStreamTagAnnotation : 'description' : is.tag.tag"
             limit="200"
             use-word-boundary="true"></truncate-long-text>
+          <span ng-if="keywords.length" ng-bind-html="imageStream | imageStreamTagAnnotation : 'description' : is.tag.tag | truncate : 1000 | highlightKeywords : keywords"></span>
```
Comment (jwforres) [inline app/views/catalog/_image.html]: so the cards may jump from showing 200 characters to showing 1000 characters?

**Comment by spadgett** [comment]
@jwforres updated

**Comment by jwforres** [comment]
[merge]

**Pull request merged**